src/template.js

```diff
@@ -19,6 +19,18 @@
  * @returns {string}  - Formatted string covering the template
 */
 
-function template(model, character, message) { }
+function template(model, character, message) {
+    let ans = model
+    if (model.indexOf(character) > -1 && message != '' ){//validating parameters
+        for (let j = 0; j < message.length; j++) {
+            ans = ans.replace(character, message[j])//replacing character in model for each letter in message
+        }
+        return ans
+    } else {
+        return "Invalid parameters"
```

Comment:
Here I would use this concept of a "guard". You want to add this return at the top of the function, then after that, you are free to assume your parameters are valid.

```
function template(model, character, message) {
  if (!(model.indexOf(character) > -1 && message != '')) return "Invalid parameters"
  /// execute the rest of the code
}